subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -158,6 +167,26 @@ private void checkProperties(Properties existingProperties) {
         }
     }
 
+    private static final String DISTRIBUTION_URL_EXCEPTION_MESSAGE = "Test of distribution url failed. Please check the values set with --gradle-distribution-url and --gradle-version.";
+
+    private void testDistributionUrl() {
+        if (distributionUrlConfigured) {
+            String url = getDistributionUrl();
+            URI uri = URI.create(url);
+            if (uri.getScheme().equals("file")) {
+                if (Files.exists(Paths.get(uri.getPath()).toAbsolutePath())) {
+                    throw new UncheckedIOException(DISTRIBUTION_URL_EXCEPTION_MESSAGE);
```

Comment:
Would be nice to add a `FileNotFoundException` to the chain here so the user has a clear idea of why the url test failed.

Reply:
Right, although `File::exists` might also return false due to missing read permissions. I wonder if `FileNotFoundException` is be misleading in this case.

Reply:
Right. My point is mostly about having the file path available somewhere in the error message.

Reply:
Got it, I'll add the url to the exception message in both cases.